app.go

```diff
@@ -638,18 +699,16 @@ func (app *App) provide(p provide) {
 	}
 
 	constructor := p.Target
-
 	switch {
 	case p.IsSupply:
 		for _, rtype := range fxreflect.ReturnTypes(constructor) {
-			fxlog.Info("supplying", fxlog.Field{Key: "type", Value: rtype}).Write(app.log)
+			entry := fxlog.Info("supplying", fxlog.Field{Key: "type", Value: rtype})
+			entry.Write(app.logBuffer)
```

Comment:
This should be reverted so that we're always writing to app.log, and app.log just points to something else.